query/src/main/java/io/keen/client/java/KeenQueryClient.java

```diff
@@ -0,0 +1,704 @@
+            throw new ServerException(response.body);
+        }
+
+        // Parse the response into a map.
+        StringReader reader = new StringReader(response.body);
+        Map<String, Object> responseMap;
+        responseMap = this.jsonHandler.readJson(reader);
+
+        // Get the result object.
+        Object result = responseMap.get(KeenQueryConstants.RESULT);
+        // for successful query, we should get a Result object. But just in case we don't...
+        if (result == null) {
+            String errorCode = responseMap.get(KeenQueryConstants.ERROR_CODE).toString();
+            String message = responseMap.get(KeenQueryConstants.MESSAGE).toString();
+
+            String errorMessage = "Error response received from server";
+            if (errorCode != null) {errorMessage += " " + errorCode;}
+            if (message != null) {errorMessage += ": " + message;}
+
+            throw new KeenQueryClientException(errorMessage);
+        }
+
+
+        return result;
+    }
+
+
+    private String formatBaseURL(String queryName) {
+        return String.format(Locale.US, "%s/%s/projects/%s/queries/%s",
+                baseUrl,
+                KeenConstants.API_VERSION,
+                project.getProjectId(),
+                queryName        // query name
+        );
+    }
+
+    private Long queryResultToLong(QueryResult result) throws KeenQueryClientException {
+        if (result.isLong()) {
+            return result.longValue();
+        } else {
+            throw new KeenQueryClientException("Count Query Error: expected Long response type.");
+        }
+    }
+
+    private Double queryResultToDouble(QueryResult result) throws KeenQueryClientException {
+        if (result.isDouble()) {
+            return result.doubleValue();
+        } else if (result.isLong()) {
+            return Long.valueOf(result.longValue()).doubleValue();
+        } else {
+            throw new KeenQueryClientException("Sum Query Error: expected Double response type.");
+        }
+    }
+
+    /**
+     * Constructs a Keen Query client using a builder.
+     *
+     * @param builder The {@link QueryBuilder} from which to retrieve this client's interfaces and settings.
+     */
+    protected KeenQueryClient(QueryBuilder builder) {
+        // Initialize final properties using the builder.
+        this.httpHandler = builder.httpHandler;
+        this.jsonHandler = builder.jsonHandler;
+        this.baseUrl = builder.baseUrl;
+        this.project = builder.project;
+    }
+
+    /**
+     * <p>
+     * Builder class for instantiating Keen Query clients.
+     * </p>
+     * <p> This builder defaults to using HttpURLConnection to handle HTTP requests.</p>
+     * <p> This builder defaults to using JacksonJsonHandler for JSON handler.</p>
+     * <p> This builder defaults to using KeenConstants.SERVER_ADDRESS for base URL.</p>
+     *
+     * @author claireyoung
+     * @since 1.0.0
+     */
+    public static class QueryBuilder {
```

Comment:
Nit: I think it would be a more idiomatic use of the builder pattern (and more consistent with the rest of the Keen SDK) to name this inner class just `Builder` instead of `QueryBuilder`. If you change it, don't forget to update the README.